src/builtin_string.cpp

```diff
@@ -949,6 +949,88 @@ static int string_split(parser_t &parser, io_streams_t &streams, int argc, wchar
     return splits.size() > arg_count ? BUILTIN_STRING_OK : BUILTIN_STRING_NONE;
 }
 
+static int string_repeat(parser_t &parser, io_streams_t &streams, int argc, wchar_t **argv) {
+	const wchar_t *short_options = L":t:q";
+	const struct woption long_options[] = {{L"times", required_argument, 0, 't'},
+										   {L"quiet", no_argument, 0, 'q'},
+										   {0, 0, 0, 0}};
+
+	bool quiet = false;
+	long times = 0;
+	wgetopter_t w;
+
+	for (;;) {
+		int opt;
+
+		if ((opt = w.wgetopt_long(argc, argv, short_options, long_options, 0)) == -1) {
+			break;
+		}
+
+		switch(opt) {
+		case 0: {
```

Comment:
This case can't happen given how you have defined the long options. You're copying a pattern found elsewhere in the fish code that is broken and needs to be changed. Specifically, neither here or anywhere else in our code do we pass a non-NULL flags value in that structure. So this is a can't happen case. I'm slowly rewriting such code. A better pattern to mimic is that of `builtin_function()` or `builtin_status()`.